tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,263 @@
+
+        // Create S3 client
+        client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(buildAWSCredentials(keyId, secret)))
+                .build();
+    }
+
+    public List<String> listFiles(String path)
+    {
+        return getObjects(path);
+    }
+
+    public List<String> listAll(String path)
+    {
+        List<String> objects = new ArrayList<String>();
+
+        List<String> pathParts = Splitter.on('/').splitToList(ensureTrailingSlashRemoved(path));
+        int pathCount = pathParts.size();
+
+        for (String object : getAllObjects(ensureTrailingSlashPresent(path))) {
+            List<String> objectParts = Splitter.on('/').splitToList(ensureTrailingSlashRemoved(object));
+            if (objectParts.size() == pathCount + 1) {
+                objects.add(object);
+            }
+        }
+
+        return objects;
+    }
+
+    public boolean exists(String path)
+    {
+        return client.doesObjectExist(bucketName, path);
+    }
+
+    public void createFolder(String path)
+    {
+        // Check to make sure directory does not already exist
+        if (exists(ensureTrailingSlashPresent(path))) {
+            return;
+        }
+
+        List<String> parts = Splitter.on('/').splitToList(path);
+        StringBuilder sb = new StringBuilder();
+
+        // Loop through all folders in path and create folders if not already exists
+        for (String objectName : parts) {
+            sb.append(ensureTrailingSlashPresent(objectName));
+            if (!exists(sb.toString())) {
+                createEmptyObject(sb.toString());
+            }
+        }
+    }
+
+    public void deleteFolder(String path)
+    {
+        // Retrieve all objects underneath this folder (including folders)
+        List<String> objects = getAllObjects(ensureTrailingSlashPresent(path));
+
+        // For all objects retrieved, delete
+        for(String objectName : objects) {
+            deleteObject(objectName);
+        }
+
+        // Delete the folder itself
+        deleteObject(ensureTrailingSlashPresent(path));
+    }
+
+    public void saveFile(String path, InputStream stream, long byteLength)
+    {
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(byteLength);
+
+        client.putObject(new PutObjectRequest(bucketName, path, stream, metadata));
+
+        logger.debug("Saved file {} to S3", path);
+    }
+
+    public void loadFile(String path, OutputStream outputStream)
+    {
+        try {
+            S3Object object = client.getObject(new GetObjectRequest(bucketName, path));
+            IOUtils.copy(object.getObjectContent(), outputStream);
+        } catch (IOException exception) {
+            throw new RuntimeException("Could not read file " + path + " from S3", exception);
+        }
+
+        logger.debug("Loaded file {} from S3", path);
+    }
+
+    public String loadFile(String path)
```

Comment:
Returning file contents (potentially binary) as `String` is bad idea. Use `ByteBuffer` or `byte[]` if method returning `OutputStream` is not enough. 
You may name this method `readFileEagerly` or sth like that to avoid naming clash with method above.